pyxem/utils/indexation_utils.py

```diff
@@ -132,6 +132,74 @@ def zero_mean_normalized_correlation(
 
     return corr_local
 
+def full_frame_correlation(image_FT, image_FT_normalization, template_coordinates, template_intensities):
+    """
+    Computes the correlation score between an image and a template in Fourier space.
+
+    Parameters:
+    -----------
+    image: numpy.ndarray
+        Intensities of the image, stored in a NxM numpy array
+    template_coordinates: numpy array
+        Array containing coordinates for non-zero intensities in the template
+    template_intensities: list
+        List of intensity values for the template.
+
+    Returns:
+    --------
+    corr_local: float
+        Correlation score between image and template.
+
+    See also:
+    ---------
+    correlate_library, fast_correlation, zero_mean_normalized_correlation
+
+    Reference:
+    ----------
+    A. Foden, D. M. Collins, A. J. Wilkinson and T. B. Britton "Indexing electron backscatter diffraction patterns with
+     a refined template matching approach" doi: https://doi.org/10.1016/j.ultramic.2019.112845
+    """
+
+    image_x = image_FT.shape[0]
+    image_y = image_FT.shape[1]
+    template = np.zeros((image_x, image_y))
+    template[template_coordinates[:, 1], template_coordinates[:, 0]] = template_intensities[:]
+
+    template_FT = np.fft.fft2(template)
+
+    #Filters can be applied here. Take user input, or standard LP/HP?
+
+    template_FT_normalization = np.sqrt(np.real(np.max(np.fft.ifft2(np.multiply(template_FT, template_FT)))))
```

Comment:
You can use `template_FT * template_FT` rather than `np.multiply()` for brevity. Or even `template_FT **2`